### DISCUSSION TITLE
SpriteMask Component

### DISCUSSION THREAD

**Discussion by Ahuang0107** [bevyengine/bevy#9762] [Ideas]
do we have any plan for adding a component for masking Sprites? Unity have SpriteMask Component that can set a texture as mask, other sprite in the same render layer will only show the overlapping parts. The effect should as follows:
![image](https://github.com/bevyengine/bevy/assets/37537012/30f2c974-e741-4933-84f4-67429b1a90b9)

i want this feature because i'm make a isometric pixel game like unpacking, I need this feature to correctly achieve the effect I want. unpacking also used SpriteMask in Unity to achieve their effect.

**Comment by superdump** [comment]
Kind of. I think 2D should also have opaque and alpha mask phases that possibly use a depth buffer, similar to 3D. With alpha mask, you could implement materials that do what you describe.
